tests/components/demo/test_lock.py

```diff
@@ -1,54 +1,49 @@
 """The tests for the Demo lock platform."""
-import unittest
+import pytest
 
 from homeassistant.components import lock
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
-from tests.common import get_test_home_assistant, mock_service
+from tests.common import async_mock_service
 from tests.components.lock import common
 
 FRONT = "lock.front_door"
 KITCHEN = "lock.kitchen_door"
 OPENABLE_LOCK = "lock.openable_lock"
 
 
-class TestLockDemo(unittest.TestCase):
-    """Test the demo lock."""
+@pytest.fixture(autouse=True)
+def setup_comp(hass):
+    """Set up demo component."""
+    hass.loop.run_until_complete(
+        async_setup_component(hass, lock.DOMAIN, {lock.DOMAIN: {"platform": "demo"}})
+    )
 
-    def setUp(self):  # pylint: disable=invalid-name
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        assert setup_component(self.hass, lock.DOMAIN, {"lock": {"platform": "demo"}})
 
-    def tearDown(self):  # pylint: disable=invalid-name
-        """Stop everything that was started."""
-        self.hass.stop()
+async def test_is_locked(hass):
+    """Test if lock is locked."""
+    assert lock.is_locked(hass, FRONT)
+    assert hass.states.is_state(FRONT, "locked")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5315232)